File: backtest-example/src/main/java/com/yuanzhixiang/bt/example/demo/StartStrategy.java
package com.yuanzhixiang.bt.example.demo;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import com.yuanzhixiang.bt.domain.model.valobj.Symbol;
import com.yuanzhixiang.bt.engine.Application;
import com.yuanzhixiang.bt.engine.Configuration;
import com.yuanzhixiang.bt.report.DefaultReporter;
import com.yuanzhixiang.bt.strategy.Strategy;

import lombok.extern.slf4j.Slf4j;

/**
 * @author yuanzhixiang
 */
@Slf4j
public class StartStrategy {

    public static void main(String[] args) {
        // 实际运行策略
        application();
    }

    public static void application() {
        // 创建配置类
        Configuration configuration = new Configuration();

        // 手续费万三
        configuration.setCommissionRate(0.0003);
        // 初始资产
        configuration.setAccountBalance(1000000);
        // 开启 T+1 交易限制
        configuration.setTodayAddOneTrade(true);
        // 注册回测报告，回测完成后会在用户 Desktop 目录下创建回测报告
        configuration.registerLifeCycle(new DefaultReporter());

        // 设置回测时间范围
        configuration.setTimeRange(
            LocalDateTime.of(2020, 1, 1, 0, 0),
            LocalDateTime.now()
        );

        // 注册需要回测的标的
        List<Symbol> symbols = new ArrayList<>();
        symbols.add(new Symbol("000001"));
        configuration.setSymbolList(symbols);

        log.info("Back test symbol count is [{}].", configuration.getSymbolList().size());

        // 创建策略类
        Strategy strategy = new MacdStrategy(configuration);

        // 运行策略
        new Application(configuration).run(strategy);
    }
}

File: backtest-example/src/main/java/com/yuanzhixiang/bt/example/demo/StartStrategy.kt
package com.yuanzhixiang.bt.example.demo

import com.yuanzhixiang.bt.service.CounterService
import com.yuanzhixiang.bt.engine.domain.Symbol
import com.yuanzhixiang.bt.engine.Configuration
import com.yuanzhixiang.bt.engine.Launcher
import com.yuanzhixiang.bt.example.report.DefaultReporter
import com.yuanzhixiang.bt.engine.Strategy
import org.slf4j.LoggerFactory
import java.time.LocalDateTime

/**
 * @author Yuan Zhixiang
 */
object StartStrategy {
    private val log = LoggerFactory.getLogger(StartStrategy::class.java)

    @JvmStatic
    fun main(args: Array<String>) {
        // 实际运行策略
        application()
    }

    private fun application() {
        // 创建配置类
        val configuration = Configuration()

        // 初始资产
        configuration.counter = CounterService(
            0.001,
            0.0006,
            0.0003,
            1000000.0,
            true
        )

        // 设置回测时间范围
        configuration.setTimeRange(
            LocalDateTime.of(2020, 1, 1, 0, 0),
            LocalDateTime.now()
        )

        // 注册需要回测的标的
        val symbols: MutableList<Symbol> = ArrayList()
        symbols.add(Symbol("000001"))
        configuration.symbolList = symbols

        // 注册回测报告，回测完成后会在用户 Desktop 目录下创建回测报告
        configuration.registerLifeCycle(
            DefaultReporter(
                configuration
            )
        )
        log.info("Back test symbol count is [{}].", configuration.symbolList.size)

        // 创建策略类
        val strategy: Strategy = MacdStrategy(configuration)

        // 运行策略
        Launcher(configuration).start(strategy = strategy)
    }
}